pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -53,21 +53,39 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, int32, err
 	})
 
 	// Build the prefix.
-	prefix := path.Join(entryPrefix, string(uuid.NewUUID()))
-
+	prefix := m.GetSecretPrefix()
+	if prefix == "" {
+		prefix = path.Join(entryPrefix, string(uuid.NewUUID()))
+	}
 	// Split the data into chunks and create the secrets on demand.
 	var err error
 	chunks := int32(0)
 	splitBytes(data, maxSecretSizeBytes, func(chunk []byte) {
 		name := fmt.Sprintf("%s-%d", prefix, chunks)
-		_, callErr := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
-			Name:         aws.String(name),
-			SecretBinary: chunk,
-			Tags:         converters.MapToSecretsManagerTags(tags),
-		})
-		if callErr != nil {
-			err = kerrors.NewAggregate([]error{callErr})
-			return
+		var done bool
+		for ok := true; ok; ok = !done {
+			_, callErr := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
+				Name:         aws.String(name),
+				SecretBinary: chunk,
+				Tags:         converters.MapToSecretsManagerTags(tags),
+			})
+			// If the secret already exists, delete and recreate
```

Comment:
Will AWS let you quickly delete & recreate a secret with the same name?

Reply:
Yes. Just ran a continuous creation/deletion loop with the SDK up to 1500.